woocommerce/woocommerce-android
# Feature/release builds on ci

This PR enables release builds on CI. 
The build is triggered by release tags. It:
- Creates the app bundle and a universal apk (signed with the shared debug key).
- Uploads the app bundle to the Play Store.
- Creates a draft release on GitHub and uploads the app bundle (only for final releases) and universal apk.

To Test:
1. Create a new branch out of this one. 
2. Update the `versionName` and `versionCode` in `build.gradle` to avoid conflicts during the upload. 
3. Commit and tag. Push the tag. 
4. Verify that the "Release Build" job is started on CircleCI. 
5. Verify that the job finishes without errors. 
6. Verify that the binary has been uploaded to the PlayStore.
7. Verify that a new GitHub release has been created and the artifact(s) uploaded. 
8. Delete the draft release on GitHub.
9. Delete the artifact on the Play Store. 

**Note:** You can find a test branch created as above here: https://github.com/woocommerce/woocommerce-android/tree/test/release-builds-on-ci, the related draft release https://github.com/woocommerce/woocommerce-android/releases/tag/untagged-2bead96efc48306a1b05 and the bundle 140 in the Play Store. 

This PR is related to the `release-toolkit` one here: https://github.com/wordpress-mobile/release-toolkit/pull/144. When everything is approved, before merging I'll create a new version of the release-toolkit and update the reference in this PR.

Update release notes:
